bgpd/bgp_clist.c

```diff
@@ -37,6 +37,37 @@
 #include "bgpd/bgp_regex.h"
 #include "bgpd/bgp_clist.h"
 
+/* Calculate new sequential number. */
+static int64_t bgp_clist_new_seq_get(struct community_list *list)
+{
+	int64_t maxseq;
+	int64_t newseq;
+	struct community_entry *entry;
+
+	maxseq = newseq = 0;
+
+	for (entry = list->head; entry; entry = entry->next) {
+		if (maxseq < entry->seq)
+			maxseq = entry->seq;
+	}
+
+	newseq = ((maxseq / 5) * 5) + 5;
```

Comment:
Nit -- the intent here is to always leave "room" for someone to insert new entries in the list which has at least some auto-generated numbers. However, it seems like if the last number was 9, the next number would be 10, failing to leave such room. Would it make sense to add an additional check such that if current largest - newly created == 1, add five more, just to avoid this?

Reply:
I think that's easy to add, but maybe we should do this for other implementations as well: access-lists, prefix-lists together?

Reply:
yes.

Reply:
At this point, should we open a new PR to fix this across all numbered lists? Maybe there should be a function that assigns the "next number" that `all the things using numbered lists` could use to get the next number?

Reply:
IMO, I would love pushing with a separate PR for all the stuff including access-lists, prefix-lists and community-lists because all of them use the same logic.